File: 2017/src/day01.js
export default function solve01(input) {
    let ans1 = 0;
    let ans2 = 0;

    let zero = "0".charCodeAt()

    for (let i = 0; i < input.length; i++) {
        if (input.charAt(i) === input.charAt((i+1) % input.length)) {
            ans1 += input.charCodeAt(i) - zero;
        }
        if (input.charAt(i) === input.charAt((i + input.length / 2) % input.length)) {
            ans2 += input.charCodeAt(i) - zero;
        }
    }

    return [ans1, ans2];
}

File: 2020/src/day01.rs
pub fn solve(input: String) -> String {
    let numbers: Vec<u32> = input.split_whitespace().map(|x| x.parse().unwrap()).collect();

    let mut ans1 = 0;
    let mut ans2 = 0;
    for i in 0..numbers.len() {
        for j in i+1..numbers.len() {
            if numbers[i] + numbers[j] == 2020 {
                ans1 = numbers[i] * numbers[j];
            }
            for k in j+1..numbers.len() {
                if numbers[i] + numbers[j] + numbers[k] == 2020 {
                    ans2 = numbers[i] * numbers[j] * numbers[k];
                }
            }
        }
    }

    format!("{}, {}", ans1, ans2)
}
